packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -149,7 +150,7 @@ class DebuggerController extends DisposableController
     var lines = _stdio.value.toList();
     final newLines = text.split('\n');
```

Comment:
What if we always text.trim() before splitting? That way we can guarantee that there will never be any leading/trailing whitespace in the incoming text, and we don't need to keep track if we've seen a newline or not?

Reply:
Hmm, I think we need to be ~faithful to the incoming chars. Esp. wrt triming leading whitespace - those may be indent that people would want to see in the output.

Similarly if we trim trailing new console output, that may not join well with the next set of chars we see.

I think what we want here logically is to stream chars into a buffer, and create new lines as we see line endings. That (with a little bit of bookkeeping) is what this code is doing.